### ISSUE TITLE
Collections: Only show New Collection button if a user can create a collection.

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#1843]
### Descriptive summary

On the collection index page, only show the New Collection button if a user can create a collection.

### Rationale

Show user a dialog if they can sometimes perform an action (e.g. can edit for some collections, but not others).  If they can never perform an action, then hide the action button (e.g. being able to add a new collection is either completely on or completely off).  

### Expected behavior

Only show New collection button if the user can create a collection.

if can? :create_any, Collection
- [ ] show New collection button

### Actual behavior

Always shows New collection button.

### Steps to reproduce the behavior

1. Create two users, user 1 and user 2
1. Login as an admin
1. Dashboard -> Settings -> Collection Types
1. Make sure user 1 is a creator for at least one collection type.
1. Make sure user 2 is not a creator for any collection types
    1. minimally Edit User Collections -> Participants and remove Registered users from Creators list
1. Login as user 1
1. Dashboard -> Collections

- [ ] Verify that New Collection button is displayed

1. Login as user 2
1. Dashboard -> Collections

- [ ] Verify that New Collection button is NOT displayed

### Related work

#1840 View and Deposit users should be able to see collection admin show page
#1841 Show/hide buttons on admin show page based on abilities
#1844 For major action buttons, should the UI hide, disable, or show dialog when user can't perform an action?

**Comment by blancoj** [comment]
@elrayle this seems to already be working.  This is the code that handles it.
https://github.com/samvera/hyrax/blob/collections-sprint/app/views/hyrax/my/collections/index.html.erb#L20

it does:  `if can? :create, Collection`
AND not what is suggested in the issue:
`if can? :create_any, Collection`

But still works.

**Comment by blancoj** [comment]
Did not mean to close issue, just to add a comment.

**Comment by elrayle** [comment]
Did you try using :create_any?  I'm not sure what rule is matching :create.  

:create_any performs a check on whether the user is a creator for at least one collection type.
```
          can :create_any, Collection if Hyrax::CollectionTypes::PermissionsService.can_create_any_collection_type?(ability: self)
```

**Comment by blancoj** [comment]
@elrayle  I just tried it with :create_any?, and it worked, so I'll send a PR with this change.

**Comment by elrayle** [comment]
Fixed by PR #2095
